File: java/src/main/java/com/sasteval/vuln/easy/CommandInjectionDirect.java
package com.sasteval.vuln.easy;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * CWE-78: OS Command Injection (Direct)
 * VULNERABILITY: User input is concatenated directly into a system command.
 */
public class CommandInjectionDirect extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        String host = request.getParameter("host");
        response.setContentType("text/plain");

        // VULN: Direct concatenation of user input into OS command
        Process process = Runtime.getRuntime().exec("ping " + host);

        try (BufferedReader reader = new BufferedReader(
                new InputStreamReader(process.getInputStream()))) {
            String line;
            while ((line = reader.readLine()) != null) {
                response.getWriter().println(line);
            }
        }
    }
}

File: dotnet/Vuln/Easy/CommandInjectionDirect.cs
using System.Diagnostics;

namespace SastEval.Vuln.Easy;

/// <summary>
/// CWE-78: OS Command Injection - User input passed directly to Process.Start.
/// </summary>
public static class CommandInjectionDirect
{
    public static IResult Handle(HttpContext context)
    {
        var host = context.Request.Query["host"].ToString();

        if (string.IsNullOrEmpty(host))
        {
            return Results.BadRequest("Missing 'host' query parameter");
        }

        // VULNERABLE: Direct concatenation of user input into OS command
        var process = Process.Start("cmd.exe", "/c ping " + host);
        process?.WaitForExit();

        return Results.Text($"Ping executed for host: {host}");
    }
}
